adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -103,29 +103,45 @@ public final void start(Future<Void> startFuture) throws Exception {
         } else {
             eventSenderSupplier = (tenant, resultHandler) -> getHonoClient().getOrCreateEventSender(tenant, resultHandler);
             telemetrySenderSupplier = (tenant, resultHandler) -> getHonoClient().getOrCreateTelemetrySender(tenant, resultHandler);
+
+            Future<Void> preStartupFuture = Future.future();
+            preStartup(preStartupFuture);
+            Future<Void> httpServerStartupFuture = Future.future();
+            preStartupFuture.compose(ar -> {
+                bindHttpServer(createRouter(), httpServerStartupFuture);
+                connectToHono(null);
+            }, httpServerStartupFuture);
             
-            Future<Void> beforeHttpStartFuture = Future.future();
-            beforeHttpServerStart(beforeHttpStartFuture);
-            beforeHttpStartFuture.setHandler(ar -> {
-               if (ar.succeeded()) {
-                   bindHttpServer(createRouter(), startFuture);
-                   connectToHono(null);
-               } else {
-                   startFuture.fail(ar.cause());
-               }
+            httpServerStartupFuture.setHandler(ar -> {
+                if (ar.succeeded()) {
+                    postStartup();
+                    startFuture.complete();
+                } else {
+                    startFuture.fail(ar.cause());
+                }
             });
         }
     }
 
     /**
      * Invoked before the Http server is started.
      * May be overridden by sub-classes to provide additional startup handling.
+     * 
+     * @param startFuture  a future that has to be completed when this operation is finished
      */
-    protected void beforeHttpServerStart(Future<Void> startFuture) {
+    protected void preStartup(final Future<Void> startFuture) {
         startFuture.complete();
     }
 
     /**
+     * Invoked after startup of this Adapter is finished.
+     * May be overridden by sub-classes.
+     */
+    protected void postStartup() {
```

Comment:
> That is why I wouldn't wanna complicate things in startup() with handling yet another Future for that and rather keep things simple. We could also rename postStartup() to onStartupSuccess() to make it clearer.

That's a great idea :+1:  I can imagine that it could be used to e.g. send out signals to other components about this event.